tests/test_solvers.py

```diff
@@ -8,31 +8,42 @@
 ############################
 
 
-def test_quadraticRoots():
+def test_rootFinder():
 
-    assert quickTest("x^2 + 2x + 1 = 0", quadraticRoots) == "(x+1.0)^(2)=0"
-    assert quickTest("x^2 + 2x = - 1", quadraticRoots) == "(x+1.0)^(2)=0"
-    assert quickTest("x^2 = - 2x - 1", quadraticRoots) == "(x+1.0)^(2)=0"
-    assert quickTest("0 = x^2 + 2x + 1", quadraticRoots) == "(x+1.0)^(2)=0"
+    #  Tests for Quardatic (2nd Degree) Equations
+    assert quickTest("x^2 + 2x + 1 = 0", rootFinder) == "(x+1.0)^(2)=0"
+    assert quickTest("x^2 + 2x = - 1", rootFinder) == "(x+1.0)^(2)=0"
+    assert quickTest("x^2 = - 2x - 1", rootFinder) == "(x+1.0)^(2)=0"
+    assert quickTest("0 = x^2 + 2x + 1", rootFinder) == "(x+1.0)^(2)=0"
 
-    assert quickTest("x^2 + 1 - 2x = 0", quadraticRoots) == "(x-1.0)^(2)=0"
-    assert quickTest("x^2 + 1 = 2x", quadraticRoots) == "(x-1.0)^(2)=0"
-    assert quickTest("x^2 = 2x - 1", quadraticRoots) == "(x-1.0)^(2)=0"
-    assert quickTest("-2x = - x^2 - 1", quadraticRoots) == "(x-1.0)^(2)=0"
-    # FIXME: assert quickTest("0 = 2x - x^2 - 1", quadraticRoots) == "(x-1.0)^(2)=0"
+    assert quickTest("x^2 + 1 - 2x = 0", rootFinder) == "(x-1.0)^(2)=0"
+    assert quickTest("x^2 + 1 = 2x", rootFinder) == "(x-1.0)^(2)=0"
+    assert quickTest("x^2 = 2x - 1", rootFinder) == "(x-1.0)^(2)=0"
+    assert quickTest("-2x = - x^2 - 1", rootFinder) == "(x-1.0)^(2)=0"
+    # FIXME: assert quickTest("0 = 2x - x^2 - 1", rootFinder) == "(x-1.0)^(2)=0"
```

Comment:
Is this still giving an error?

Reply:
Yes, it is (kind of stuck in a loop when this input is given). Will update if it is resolved with next updates.